### PR TITLE
Add coverage for Docker CLI: DockerActivationKeyTestCase and DockerComputeResourceTestCase tests

### PR DESCRIPTION
**Pull request by abalakh**
Implemented `DockerActivationKeyTestCase` and `DockerComputeResourceTestCase` stubs.
As usual, updated corresponding API tests.
Part of #2705

### CODE REVIEW

**Comment by abalakh** [comment]
Tests results:
CLI:
```
nosetests tests/foreman/cli/test_docker.py:DockerComputeResourceTestCase
....................
----------------------------------------------------------------------
Ran 20 tests in 189.635s

OK
nosetests tests/foreman/cli/test_docker.py:DockerActivationKeyTestCase 
....
----------------------------------------------------------------------
Ran 4 tests in 327.387s

OK
```

API:
```
nosetests tests/foreman/api/test_docker.py -m test_create_one_docker_repo
.......
----------------------------------------------------------------------
Ran 7 tests in 80.296s

OK
 nosetests tests/foreman/api/test_docker.py -m test_update_docker_repo_name
.......
----------------------------------------------------------------------
Ran 7 tests in 104.791s

OK
nosetests tests/foreman/api/test_docker.py:DockerActivationKeyTestCase    
....
----------------------------------------------------------------------
Ran 4 tests in 171.254s

OK
nosetests tests/foreman/api/test_docker.py:DockerComputeResourceTestCase
....................
----------------------------------------------------------------------
Ran 20 tests in 117.685s

OK
```

File: robottelo/cli/factory.py [hunk 1]
```diff
@@ -243,7 +243,7 @@ def make_container(options=None):
         u'location-ids': None,
         u'locations': None,
         u'memory': None,
-        u'name': gen_string('utf8'),
+        u'name': gen_string('alphanumeric'),
```
Comment (abalakh) [inline robottelo/cli/factory.py]: Bugfix. As error message says,
```
Invalid container name (���), only [a-zA-Z0-9][a-zA-Z0-9_.-] are allowed
```

File: tests/foreman/cli/test_docker.py [hunk 2]
```diff
@@ -998,18 +1157,9 @@ class DockerComputeResourceTestCase(CLITestCase):
...
-    @run_only_on('sat')
-    @data(
-        gen_string('alpha'),
-        gen_string('alphanumeric'),
-        gen_string('numeric'),
-        gen_string('latin1'),
-        gen_string('utf8'),
-        gen_string('html'),
-    )
+    @data(*valid_data_list())
```
Comment (abalakh) [inline tests/foreman/cli/test_docker.py]: `test_create_internal_docker_compute_resource` and `test_create_external_docker_compute_resource` are doing basically the same for different urls. does anyone have ideas how to correctly merge 2 lists of different length in data decorator? i was thinking about smth like 
```python
@data(zip(valid_data_list(), cycle([INTERNAL_DOCKER_URL, EXTERNAL_DOCKER_URL]))
```
but it doesn't look readable :) and needs `itertools` to be imported
of course, i could do smth like
```python
          {'name': gen_string('alphanumeric', randint(1, 255)),
           'url': INTERNAL_DOCKER_URL},
          {'name': gen_string('alpha', randint(1, 255)),
           'url': INTERNAL_DOCKER_URL},
          {'name': gen_string('cjk', randint(1, 85)),
           'url': INTERNAL_DOCKER_URL},
          {'name': gen_string('latin1', randint(1, 255)),
           'url': INTERNAL_DOCKER_URL},
          {'name': gen_string('numeric', randint(1, 255)),
           'url': INTERNAL_DOCKER_URL},
          {'name': gen_string('utf8', randint(1, 85)),
           'url': INTERNAL_DOCKER_URL},
          {'name': gen_string('html', randint(1, 85)),
           'url': INTERNAL_DOCKER_URL},
          {'name': gen_string('alphanumeric', randint(1, 255)),
           'url': EXTERNAL_DOCKER_URL},
          {'name': gen_string('alpha', randint(1, 255)),
           'url': EXTERNAL_DOCKER_URL},
          {'name': gen_string('cjk', randint(1, 85)),
           'url': EXTERNAL_DOCKER_URL},
          {'name': gen_string('latin1', randint(1, 255)),
           'url': EXTERNAL_DOCKER_URL},
          {'name': gen_string('numeric', randint(1, 255)),
           'url': EXTERNAL_DOCKER_URL},
          {'name': gen_string('utf8', randint(1, 85)),
           'url': EXTERNAL_DOCKER_URL},
          {'name': gen_string('html', randint(1, 85)),
           'url': EXTERNAL_DOCKER_URL},)
```
but this doesn't look like the best solution either. is there a better way? or maybe i should leave 2 tests as they are?

Comment (Ichimonji10) [inline tests/foreman/cli/test_docker.py]: I think it's fine to use itertools in a test. itertools is in the standard library, and we use it elsewhere, so it's not as if you're pulling in some exotic dependency.

I wouldn't stress too much about making making the code in the `@data(...)` decorator look nice. We're going to drop DDT, and when we do that, that'll be a better time to be inventive and come up with good ways to gernerate data.

File: tests/foreman/api/test_docker.py [hunk 3]
```diff
@@ -1063,55 +1038,6 @@ def test_create_external_docker_compute_resource(self, name):
         self.assertEqual(compute_resource.provider, DOCKER_PROVIDER)
         self.assertEqual(compute_resource.url, EXTERNAL_DOCKER_URL)
 
-    def test_update_external_docker_compute_resource(self):
```
Comment (Ichimonji10) [inline tests/foreman/api/test_docker.py]: :+1: Thanks for removing this redundancy.

**Comment by Ichimonji10** [comment]
ACK

File: robottelo/cli/factory.py (hunk 1 above)
Comment (elyezer) [inline robottelo/cli/factory.py]: :+1:

File: tests/foreman/cli/test_docker.py (hunk 2 above)
Comment (elyezer) [inline tests/foreman/cli/test_docker.py]: cycle seems to be the right and cleaner solution to me.

**Comment by elyezer** [comment]
Will merge as is and if needed update the `@data` to use the cycle and merge both tests.

**Comment by elyezer** [comment]
ACK

**Pull request merged**